### ISSUE TITLE
Suggestion to accept stack access with no-unused-expressions

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#10733]
**What rule do you want to change?**

no-unused-expressions

**Does this change cause the rule to produce more or fewer warnings?**

Less

**How will the change be implemented? (New option, new default behavior, etc.)?**

I have no strong opinion about either way.

**Please provide some example code that this change will affect:**

```js
try {
  throw new Error('foo')
} catch (err) {
  err.stack
}
```

**What does the rule currently do for this code?**

It reports an error. This is absolutely correct but V8 lazily formats the stack trace on access.

**What will the rule do after it's changed?**

It does not report errors anymore. I have no strong opinion about this one but I had to use multiple ignore statements or could not use the rule at all in some cases.

**Comment by platinumazure** [comment]
I'm confused. Why would you not do `console.log(err.stack);`?

**Comment by BridgeAR** [comment]
It is required in combination with `Error.prepareStackTrace`. See https://github.com/nodejs/node/blob/df08779e0dfa82/test/common/index.js#L525 for an example.

**Comment by platinumazure** [comment]
Honestly, I'm not sure this is an enhancement we should do. We generally don't like to do runtime- or framework-specific enhancements to core rules. I think this sort of odd exceptional case is what [ESLint disable comments](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments) are for.

```js
try {
  throw new Error('foo')
} catch (err) {
  err.stack     // eslint-disable-line no-unused-expressions
}
```
